### PR TITLE
Add ConnectionClosedException as transient error

### LINKED ISSUE #16642: Retry connection connection closed during resolution.
Currently, some error types are marked as [network issues](https://github.com/gradle/gradle/blob/master/subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/transport/NetworkingIssueVerifier.java). I'd like to include `org.apache.http.ConnectionClosedException`.

### Expected Behavior
org.apache.http.ConnectionClosedException retries

### Current Behavior
org.apache.http.ConnectionClosedException fails

### Context
We have some rather large jars that need to get downloaded. We have a lot of CI builds. Very occasionally, the jar download fails in the middle, causing an unnecessary build failure.

### PR DESCRIPTION
**Pull request by hegyibalint**
Fixes #16642

TL;DR: By adding Apache's ConnectionClosedException, when the exception occurs, the build doesn't fail but instead retries the request.

### CODE REVIEW

**Comment by hegyibalint** [comment]
@bot-gradle test RFM

**Review by ljacomet — APPROVED** [review]

**Comment by hegyibalint** [comment]
@bot-gradle test and merge

**Comment by blindpirate** [comment]
@bot-gradle test and merge

**Comment by blindpirate** [comment]
@bot-gradle test and merge
